### PR TITLE
Feature/11264

### PR DESCRIPTION
**Pull request by shabeeb-aot**
*Issue #:* /bcgov/entity#11264

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

**Comment by shabeeb-aot** [comment]
Did not added any test case now, Will add in next PR while adding all components

File: src/App.vue
Comment (shabeeb-aot) [inline src/App.vue:391]: can move to store if we have subtitle for other types

File: src/App.vue
Comment (shabeeb-aot) [inline src/App.vue:660]: overriding filing type as SP for testing purposes. will remove once create is in dev

File: src/resources/Dissolutions/soloPropCompany.ts
Comment (shabeeb-aot) [inline src/resources/Dissolutions/soloPropCompany.ts:17]: All these are needed to component work even we are not explicitly using it. When I am removing these its giving error.

**Comment by severinbeauvais** [comment]
The linting and build-check CI steps must pass before this PR is merged. (I'm OK with testing-coverage aka unit tests coming in another ticket.)

Also, you should update the app version in package.json (and then run `npm i`) so that we can track app versions through the environment, eg, click on "i" icon at bottom right of UI:

image

File: src/App.vue
Comment (severinbeauvais) [inline src/App.vue:101]: In my experiene, it's easier to lay out whitespace if you only use top whitespace. (That way, each element only has to be concerned with the whitespace above it.) (There is an exception for the bottom element in some blocks.)

I believe the bottom margin here, and the top margin in the Stepper component below, will create too much whitespace.

If the intent is to create whitespace between the subtitle and the Signin component (assuming the Stepper is not rendered) then that should be handled either in the Stepper or in the Signin. This is not the subtitle's problem to worry about, and doing so complicates the layout and logic.

Comment (severinbeauvais) [reply src/App.vue:101]: Sorry, the Signin component doesn't render anything. I meant whatever component displays below the Stepper.

File: src/enums/routeNames.ts
Comment (severinbeauvais) [inline src/enums/routeNames.ts:30]: The first part of this name conflicts with the other dissolution filings above.
The second part of this name is too general: every business is an entity.

If this feature specific to Firms? Will it apply to other corp types -- which ones? Will the new ones use the same route / code?

I suggest, maybe, "firm-dissolution" ...?

If you change this, please sync the component name as well.

File: src/resources/Dissolutions/index.ts
Comment (severinbeauvais) [inline src/resources/Dissolutions/index.ts:7]: Our style for "const" objects like this one is to PascalCase them. This is so they look like unmutable things instead of variables.

File: src/resources/Dissolutions/soloPropCompany.ts
Comment (severinbeauvais) [reply src/resources/Dissolutions/soloPropCompany.ts:17]: Which components are giving you an error? They will need to be fixed to allow for this data to not exist. Having dummy data here is not a good solution. (It looks like it does something, but it actually does nothing, leading to confusion and increased code maintenance cost.)

File: src/store/getters/state-getters.ts
Comment (severinbeauvais) [inline src/store/getters/state-getters.ts:68]: You should use `isTypeSoleProp(state)` instead.

File: src/enums/filingNames.ts
Comment (cameron-freshworks) [inline src/enums/filingNames.ts:5]: Perhaps this could be simplified to `DISSOLUTION`?

File: src/resources/Dissolutions/soloPropCompany.ts
Comment (cameron-freshworks) [reply src/resources/Dissolutions/soloPropCompany.ts:1]: I like this. We should name all our components or ts files after a star wars characters :)
Will be challenging!

File: src/resources/Dissolutions/soloPropCompany.ts
Comment (cameron-freshworks) [reply src/resources/Dissolutions/soloPropCompany.ts:17]: Is it the components giving errors or are they type errors from not adhering to the interface?

For example , `steps` i know is optional in your flow, so it might make sense to create an interface unique to Firms or make this property optional in the IF. 

If the process is failing, it's certainly worth looking into, as a lot of the current components rely on this templated data. |
If there is no text content difference between firms and corporations, then this resourcing could potentially feel redundant, but we've found this easier to make changes after the fact :)

But yes, you might not require all of this, and if so, please remove and allow optional values in the IF or perhaps a unique firm dissolutionIF.

File: src/router/routes.ts
Comment (cameron-freshworks) [inline src/router/routes.ts:164]: When you say `entity` are we referring to firms specifically? 
If so, perhaps we should rename these references to indicate that, as entity is very high level, meaning it could mean any single type of business, corporation firm etc

File: src/router/routes.ts
Comment (cameron-freshworks) [reply src/router/routes.ts:171]: Will this default to false if a user find themselves on another route? 

To this point, we've always protected the routes between Dissolutions and Incorporations, not allowing temp reg numbers to access these routes, and vice versa. 

Might be worth looking into that mechanism in order to prevent Firms or Corporations from accessing each others routes, due to the fact they might?* share the same filing type (which i think we've used to differentiate in the past).

File: src/views/DissolutionEntity/DissolutionEntity.vue
Comment (cameron-freshworks) [inline src/views/DissolutionEntity/DissolutionEntity.vue:375]: Same as above

File: src/enums/filingNames.ts
Comment (severinbeauvais) [reply src/enums/filingNames.ts:5]: I like the consistency with the route name / component name.

On the other hand, this is just used for display purposes, right? So exactly what is this filing type code? Then make the enum key the same.

File: src/router/routes.ts
Comment (severinbeauvais) [reply src/router/routes.ts:171]: It will be undefined (which is falsy) if it's not specified in the meta for another route. I'm OK with that.

I agree on the protection but it's actually pretty loose... in App.vue the code takes one path for Dissolutions and another page for IAs/Registrations. I don't have a better way in mind atm.

File: src/App.vue
Comment (severinbeauvais) [reply src/App.vue]: Obsolete review comment? If so please delete it.

File: src/views/DissolutionEntity/DissolutionEntity.vue
Comment (severinbeauvais) [reply src/views/DissolutionEntity/DissolutionEntity.vue:369]: PS If you recently rebased to the latest main branch then you already have most of the latest shared components available to you (and I updated the local ones).

File: src/views/DissolutionEntity/DissolutionEntity.vue
Comment (severinbeauvais) [reply src/views/DissolutionEntity/DissolutionEntity.vue:375]: Yes, mostly we how use v-cards with no extra CSS (ie, `<v-card flat>`) (although probably some top padding for whitespace).

File: src/router/routes.ts
Comment (shabeeb-aot) [reply src/router/routes.ts:171]: Sure, I will check that 580, Currently, it's working when the filing type is 'SP'. Once we have the actual value I will refactor it, if needed.
Also from app.vue data will show according to filing type.

File: src/store/getters/state-getters.ts
Comment (shabeeb-aot) [reply src/store/getters/state-getters.ts:68]: Thank you , I never know we had a getter for it 👍

File: src/enums/filingNames.ts
Comment (shabeeb-aot) [reply src/enums/filingNames.ts:5]: changed to DISSOLUTION_FIRM, to consistences with file name , is it okey?

File: src/router/routes.ts
Comment (shabeeb-aot) [reply src/router/routes.ts:164]: changed to dissolution-firm as name consistences with file/ component name. if we have multiple routes for Sp and GP we can use the same name for routes

File: src/resources/Dissolutions/soloPropCompany.ts
Comment (shabeeb-aot) [reply src/resources/Dissolutions/soloPropCompany.ts]: , I was getting error when I remove ```reviewAndConfirm``` section , actually, there is one section  ```completingPartyStatement``` was used inside Certify component

File: src/views/DissolutionEntity/DissolutionEntity.vue
Comment (shabeeb-aot) [reply src/views/DissolutionEntity/DissolutionEntity.vue]: I removed some codes, when i am removing the above code, its style is missing.
I will check this later on next PR
Screen Shot 2022-03-16 at 12 18 59 PM

File: src/resources/Dissolutions/stepTemplates.ts
Comment (severinbeauvais) [reply src/resources/Dissolutions/stepTemplates.ts:89]: Now, you use a route meta prop to disable the stepper, right?

So I think this comment is obsolete. Please delete it.

File: src/resources/Dissolutions/index.ts
Comment (severinbeauvais) [inline src/resources/Dissolutions/index.ts:7]: To be consistent with the other SP resource (in registrations), please name the file "soleProprietorship".

File: src/router/routes.ts
Comment (severinbeauvais) [reply src/router/routes.ts]: LGTM

File: package.json
Comment (severinbeauvais) [inline package.json:3]: You need to update line 77 in this file from `git:` to `https:`.

File: src/resources/Dissolutions/stepTemplates.ts
Comment (shabeeb-aot) [reply src/resources/Dissolutions/stepTemplates.ts:89]: yeah, But still without a stepper app will not work, I will remove last last comment

File: package.json
Comment (shabeeb-aot) [reply package.json:3]: Updated

File: src/enums/filingNames.ts
Comment (severinbeauvais) [inline src/enums/filingNames.ts:9]: I'm OK with declaring a new, related enum in the same file.

However, this message seems specific to a dissolution, and I don't get that from its key. I don't see where this is used so I can't form an opinion on how this string should be stored.

Also, I think this line is too long. Check the linting results.

File: src/resources/Dissolutions/soloPropCompany.ts
Comment (severinbeauvais) [reply src/resources/Dissolutions/soloPropCompany.ts]: This resources file looks good now. (Only contains the objects it needs 👍 )

File: src/views/DissolutionEntity/DissolutionEntity.vue
Comment (cameron-freshworks) [reply src/views/DissolutionEntity/DissolutionEntity.vue]: 🤔  Hmm, yes. 
Perhaps you will either need to rebase as Severin said to pull in the stylings that were re applied locally after the Shared Component Updates or might need to wrap them locally if they need it.

File: src/resources/Dissolutions/soloPropCompany.ts
Comment (severinbeauvais) [reply src/resources/Dissolutions/soloPropCompany.ts]: @cameron-freshworks Can we resolve this conversation?

File: src/views/DissolutionEntity/DissolutionEntity.vue
Comment (severinbeauvais) [reply src/views/DissolutionEntity/DissolutionEntity.vue]: No, that's perfect.

It means your page needs to specify the padding on these components. (The components don't specify their own padding.)

Try `class="py-8 px-6"` to be consistent with registration filings, or match your designs.

File: src/resources/Dissolutions/soloPropCompany.ts
Comment (cameron-freshworks) [reply src/resources/Dissolutions/soloPropCompany.ts]: @severinbeauvais You bet!

File: src/resources/Dissolutions/stepTemplates.ts
Comment (severinbeauvais) [reply src/resources/Dissolutions/stepTemplates.ts]: So you need a single step for, say, the actions to work? But you also want to hide the stepper?

File: src/views/DissolutionEntity/DissolutionEntity.vue
Comment (cameron-freshworks) [reply src/views/DissolutionEntity/DissolutionEntity.vue]: Will resolve assuming your adding in the styling then! 👍

Comment (cameron-freshworks) [reply src/views/DissolutionEntity/DissolutionEntity.vue]: Maybe just post a pic :)

File: src/store/getters/state-getters.ts
Comment (severinbeauvais) [inline src/store/getters/state-getters.ts:658]: This does not apply to all, say, SP filings. If we merge this code then SP registration filings will incorrectly display this subtitle.

So the filing type needs to be taken into account.

File: src/enums/filingNames.ts
Comment (severinbeauvais) [reply src/enums/filingNames.ts:9]: OK, I saw your new getter.

I left a comment on your getter. I think this needs a different solution, so I'm resolving this conversation and will look at your new solution.

File: src/App.vue
Comment (severinbeauvais) [inline src/App.vue:388]: Please add return type.

File: src/resources/Dissolutions/soleProprietorship.ts
Comment (severinbeauvais) [inline src/resources/Dissolutions/soleProprietorship.ts:23]: Please verify that this is correct Act for firms.

File: src/resources/Dissolutions/soleProprietorship.ts
Comment (severinbeauvais) [inline src/resources/Dissolutions/soleProprietorship.ts:9]: Incorrect display name.

Also below on line 12.

File: src/resources/Dissolutions/stepTemplates.ts
Comment (severinbeauvais) [inline src/resources/Dissolutions/stepTemplates.ts:88]: Please update these comments. I'd like to see:
- the correct step name in the comments
- the correct entity types this applies to
- a comment that a single step is needed even though the stepper is not used for firm dissolution filings

Comment (severinbeauvais) [reply src/resources/Dissolutions/stepTemplates.ts:88]: Also please rename this object to match the type of filing this applies to.

File: src/router/routes.ts
Comment (severinbeauvais) [reply src/router/routes.ts:171]: Just make sure what you do works for SP dissolutions and doesn't interfere with SP registrations.

Faiz, you can resolve this conversation once you have read this comment 😀

File: src/enums/routeNames.ts
Comment (severinbeauvais) [inline src/enums/routeNames.ts:30]: Please update the comment.

The name still clashes with the "other" dissolution filings, but since we don't know how this will change in the future, I think the name is OK.

Also please add a space before the `=`.

File: src/resources/Dissolutions/stepTemplates.ts
Comment (severinbeauvais) [reply src/resources/Dissolutions/stepTemplates.ts]: Resolving this conversation. I get it now. See other comment in this file.

File: src/router/routes.ts
Comment (severinbeauvais) [reply src/router/routes.ts]: @cameron-freshworks OK to resolve this conversation?

File: src/resources/Dissolutions/soleProprietorship.ts
Comment (severinbeauvais) [reply src/resources/Dissolutions/soleProprietorship.ts:30]: Faiz, you can resolve this conversation after you read the comments.

File: src/views/DissolutionFirm/DissolutionFirm.vue
Comment (severinbeauvais) [inline src/views/DissolutionFirm/DissolutionFirm.vue:66]: Add a blank line between sections. This makes it nice to look at when using code folding feature in IDE. (Also nice to review :) )

File: src/router/routes.ts
Comment (cameron-freshworks) [reply src/router/routes.ts]: You bet, thanks for the update!

File: src/views/DissolutionFirm/DissolutionFirm.vue
Comment (severinbeauvais) [inline src/views/DissolutionFirm/DissolutionFirm.vue:112]: Please ask BA/PO if this section applies to SBC Staff also.

Re: https://github.com/bcgov/business-create-ui/blob/main/src/store/getters/state-getters.ts#L93

Comment (severinbeauvais) [reply src/views/DissolutionFirm/DissolutionFirm.vue:112]: This might also apply to the Staff Payment section.

**Review by cameron-freshworks — APPROVED** [review]
Hard to jump into new code, gj! 
LGTM, pending resolving the CI checks and Severins comments 👍

File: src/views/DissolutionFirm/DissolutionFirm.vue
Comment (severinbeauvais) [inline src/views/DissolutionFirm/DissolutionFirm.vue:271]: You probably don't need this, or the `::v-deep` on line 274.

File: src/views/DissolutionFirm/DissolutionFirm.vue
Comment (severinbeauvais) [inline src/views/DissolutionFirm/DissolutionFirm.vue:311]: I'm pretty sure you don't need inner-col-1 or inner-col-2.

File: src/views/DissolutionFirm/DissolutionFirm.vue
Comment (severinbeauvais) [reply src/views/DissolutionFirm/DissolutionFirm.vue:271]: You probably don't need ANY of the v-deep style overrides in here.

File: package-lock.json
Comment (severinbeauvais) [inline package-lock.json:3]: The reason that the CI checks are still failing is that you updated the "vue-test-utils-helpers" import in the package.json file but you did not run `npm i` afterwards (and/or you did not commit the updated package-lock.json file).

The build server uses THIS file to determine what packages to install (and what version). This way, the build server will use exactly the same packages to build the app that you used locally.

File: src/views/DissolutionFirm/DissolutionFirm.vue
Comment (severinbeauvais) [reply src/views/DissolutionFirm/DissolutionFirm.vue:112]: OK.

(I saw todays that there is a Design ticket to look at changes for staff filings in Create UI... so it's coming.)

File: src/resources/Dissolutions/soleProprietorship.ts
Comment (shabeeb-aot) [reply src/resources/Dissolutions/soleProprietorship.ts:30]: I didn't understand this comment exactly

Comment (severinbeauvais) [reply src/resources/Dissolutions/soleProprietorship.ts:30]: This was just a note for you. I see you've read it so I will resolve this conversation.

File: src/views/DissolutionFirm/DissolutionFirm.vue
Comment (severinbeauvais) [reply src/views/DissolutionFirm/DissolutionFirm.vue]: I see you removed the `::v-deep` selector modifier... Sorry, this is not what I meant. I meant some of these CSS blocks may not be needed any more (because the shared components were updated).

Did you check this? Do you still need this CSS here? If so, let me know and I'll resolve this conversation.

File: src/store/getters/state-getters.ts
Comment (severinbeauvais) [inline src/store/getters/state-getters.ts:656]: You can use getter `isDissolutionFiling` for the first statement.

File: src/views/DissolutionFirm/DissolutionFirm.vue
Comment (shabeeb-aot) [reply src/views/DissolutionFirm/DissolutionFirm.vue]: Yeah, we still need this CSS, else the design will completely change

Screen Shot 2022-03-17 at 10 47 56 AM

**Review by severinbeauvais — APPROVED** [review]
Looks good. You can wait for Saravan's approval or you can merge it now. Up to you.

**Pull request merged**
